src/mlpack/methods/gmm/em_fit_impl.hpp

```diff
@@ -231,16 +283,28 @@ InitialClustering(const arma::mat& observations,
   // Run clustering algorithm.
   clusterer.Cluster(observations, dists.size(), assignments);
 
+  // Check if the type of Distribution is DiagonalGaussianDistribution.  If so,
+  // we can get faster performance by using diagonal elements when calculating
+  // the covariance.
+  const bool isDiagGaussDist = std::is_same<Distribution,
+      distribution::DiagonalGaussianDistribution>::value;
+
   std::vector<arma::vec> means(dists.size());
-  std::vector<arma::mat> covs(dists.size());
+
+  // Conditional covariance instantiation.
+  std::vector<typename std::conditional<isDiagGaussDist,
+      arma::vec, arma::mat>::type> covs(dists.size());
 
   // Now calculate the means, covariances, and weights.
   weights.zeros();
   for (size_t i = 0; i < dists.size(); ++i)
   {
     means[i].zeros(dists[i].Mean().n_elem);
-    covs[i].zeros(dists[i].Covariance().n_rows,
-                  dists[i].Covariance().n_cols);
+    if (isDiagGaussDist)
+      covs[i].zeros(dists[i].Covariance().n_elem);
+    else
+      covs[i].zeros(dists[i].Covariance().n_rows,
```

Comment:
Do you mind to wrap this with ``{}`` to improve the readability?

Reply:
Of course not, I'll wrap this with `{}`